test/integration/image-component/custom-resolver/pages/_app.js

```diff
@@ -0,0 +1,11 @@
+import { registerCustomImageLoader } from 'next/image'
+
+registerCustomImageLoader(({ src, width, quality }) => {
+  return `https://customresolver.com/${src}?w~~${width},q~~${quality}`
```

Comment:
Just to point out that, for security reasons, I suggest to use domains that you can control, even for integration tests. At least one can use example.com for the purpose. 😄